### ISSUE TITLE
Apple AirPlay causes GLFW error in glfwInit ("Cocoa: Failed to find service port for display")

### ISSUE THREAD

**Issue by JannikGM** [glfw/glfw#1791]
I'm getting GLFW errors when my MacBook (macOS 10.15.7) is connected via [AirPlay](https://en.wikipedia.org/wiki/AirPlay). I'm new to macOS, so I'm not 100% sure about the package version I'm using, but I appear to be running on GLFW 3.3.2 (installed via brew).

Naively written applications will probably exit or trigger into a debugger when they get an GLFW error callback. For example, the mapbox GLFW platform sample:

```
Current executable set to '/tmp/mapbox-gl-native/build/platform/glfw/mbgl-glfw' (x86_64).
(lldb) r
Process 84490 launched: '/tmp/mapbox-gl-native/build/platform/glfw/mbgl-glfw' (x86_64)
Assertion failed: (false), function glfwError, file ../platform/glfw/glfw_view.cpp, line 112.
Process 84490 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = hit program assert
    frame #4: 0x000000010004bfb7 mbgl-glfw`glfwError(error=65544, description="Cocoa: Failed to find service port for display") at glfw_view.cpp:112:5
   109
   110 	void glfwError(int error, const char *description) {
   111 	    mbgl::Log::Error(mbgl::Event::OpenGL, "GLFW error (%i): %s", error, description);
-> 112 	    assert(false);
   113 	}
   114
   115 	GLFWView::GLFWView(bool fullscreen_, bool benchmark_, const mbgl::ResourceOptions &options)
Target 1: (mbgl-glfw) stopped.
```

Here is the backtrace for that situation (unfortunately, I don't have GLFW symbols):

```
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = hit program assert
    frame #0: 0x00007fff69d3533a libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff69df1e60 libsystem_pthread.dylib`pthread_kill + 430
    frame #2: 0x00007fff69cbc808 libsystem_c.dylib`abort + 120
    frame #3: 0x00007fff69cbbac6 libsystem_c.dylib`__assert_rtn + 314
  * frame #4: 0x000000010004bfb7 mbgl-glfw`glfwError(error=65544, description="Cocoa: Failed to find service port for display") at glfw_view.cpp:112:5
    frame #5: 0x00000001061d182e libglfw.3.dylib`_glfwInputError + 491
    frame #6: 0x00000001061d8244 libglfw.3.dylib`_glfwPollMonitorsNS + 733
    frame #7: 0x00000001061d7243 libglfw.3.dylib`_glfwPlatformInit + 1649
    frame #8: 0x00000001061d1bb6 libglfw.3.dylib`glfwInit + 74
    frame #9: 0x000000010004c34f mbgl-glfw`GLFWView::GLFWView(this=0x00007ffeefbfedd0, fullscreen_=false, benchmark_=false, options=0x00007ffeefbfd748) at glfw_view.cpp:124:10
    frame #10: 0x000000010005142c mbgl-glfw`GLFWView::GLFWView(this=0x00007ffeefbfedd0, fullscreen_=false, benchmark_=false, options=0x00007ffeefbfd748) at glfw_view.cpp:119:43
    frame #11: 0x000000010000ba17 mbgl-glfw`main(argc=1, argv=0x00007ffeefbff848) at main.cpp:107:14
    frame #12: 0x00007fff69bedcc9 libdyld.dylib`start + 1
    frame #13: 0x00007fff69bedcc9 libdyld.dylib`start + 1
```

I compared it to the code, and I assume it's this:

https://github.com/glfw/glfw/blob/a2674a903434b7dfc0b0bcc5d0b479da417367ff/src/cocoa_monitor.m#L87-L92

Ignoring that error will not cause any issues (or none that I noticed). The GLFW window will behave normally on the native display and the AirPlay display / mirror.

This case should be handled more gracefully. If there's an option to check for AirPlay, a respective name should be returned (ideally, the name of the AirPlay display?).

**Comment by elmindreda** [comment]
This should be fixed now with 955fbd9d265fa95adf9cb94896eb9a516aa50420. Thank you for the detailed bug report!

**Comment by xbbkok** [comment]
I also meet the problem
